lib/auth/auth_with_roles.go

```diff
@@ -854,15 +854,17 @@ func (a *AuthWithRoles) Ping(ctx context.Context) (proto.PingResponse, error) {
 	}, nil
 }
 
+type contextKey string
+
 // withUpdateBy creates a child context with the updated_by value set.
 // Helps capture the user who modified a resource.
 func withUpdateBy(ctx context.Context, user string) context.Context {
-	return context.WithValue(ctx, events.UpdatedBy, user)
+	return context.WithValue(ctx, contextKey(events.UpdatedBy), user)
```

Comment:
Is this to ensure the `events.UpdatedBy` is a string?

Reply:
Nope, it's to prevent other packages from re-using the same context key: https://staticcheck.io/docs/checks#SA1029

With this change, even if another package used `"updated_by"` as a context key, it wouldn't match this key due to different types.